engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/DebugLocalScope.java

```diff
@@ -188,15 +193,56 @@ boolean isMemberReadable(String memberName) {
   @ExportMessage
   @TruffleBoundary
   Object readMember(String member, @CachedLibrary("this") InteropLibrary interop)
-      throws UnknownIdentifierException {
+      throws UnknownIdentifierException, UnsupportedMessageException {
     if (!allBindings.containsKey(member)) {
       throw UnknownIdentifierException.create(member);
     }
     FramePointer framePtr = allBindings.get(member);
     var value = getValue(frame, framePtr);
+    if (value != null) {
+      if (isHostObject(value)) {
+        // Determine if the host object can be converted to corresponding Enso value.
+        var interopUncached = InteropLibrary.getUncached();
+        var firstMember = readFirstMember(value);
+        if (firstMember != null) {
+          var builtinsScope = EnsoContext.get(interop).getBuiltins().getScope();
+          var symbol = UnresolvedSymbol.build(firstMember, builtinsScope);
+          var polyglotCallType =
+              HostMethodCallNode.getPolyglotCallType(value, symbol, interopUncached);
+          try {
+            switch (polyglotCallType) {
+              case CONVERT_TO_DATE -> {
```

Comment:
Here is the main idea. Chrome inspector instrument builds `DebugLocalScope` which holds a materialized frame with all the bindings. If a binding is a host object that can/should be converted to an appropriate Enso builtin type before sending back to the inspector, it is converted here. This is pretty much the same kind of conversion that is done inside [InvokeMethodNode](https://github.com/enso-org/enso/blob/8e765953a4d6caed716f5ed536a00f527cb7ecd6/engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java) on the `self` argument before any kind of method dispatch is done.